blog/2024-01-23-state-machines-whats-in-a-name/index.mdx

```diff
@@ -0,0 +1,186 @@
+
+Fast forward to 2024, with the [recent release of XState v5](https://stately.ai/blog/2023-12-01-xstate-v5), one particular style convention for event names is strongly encouraged and that's **`dot.case`**. If you recall in the image of the Quartz watch, above, there were events similar to **`battery.inserted`**, **`battery.removed`**, and **`weak battery.dies`**. This style allows for a [new feature in XState v5 called wildcard transitions](https://stately.ai/docs/transitions#wildcard-transitions). For example, if you had an action called `logBatteryStatus` you could listen to that event with a wildcard transition like this:
+
+```typescript
+states: {
+  Running: {
+    on: {
+      'battery.*': {
+        actions: 'logBatteryStatus',
+      },
+    },
+  }
+}
+```
+
+I personally like this convention for events because it visually distinguishes them from other name in the state machine and its lack of heavy capitalization make it look more **`streamlined.for.action`**, like events should be!
+
+:::warning
+
+Unlike for event names, it is currently not possible to include a `.` when naming states or machines. Please do not do this.
+
+:::
+
+## Other tips
+
+Here are some additional considerations to think about when choosing a naming style for use with XState to make your state machines more readable and easier to understand.
+
+### Improved readability
+
+One of my favorite tips for state machine naming is to use contrasting styles. Contrast is a key ingredient for many things we love from food to color, art, and music. The readability of state machines can also benefit from contrast. If you use **`SHOUTY_UPPERCASE`** event names the please don’t also use that case for state names or… anything else in the same machine really. At the same time, don’t just name every state, event, and action in **`plain old lower case`** unless you want your state machine to be read like the equivalent of a bored teen with “vocal fry”. Differentiation can help readers distinguish certain elements of the state machine from others and this applies to readability on the editor canvas as well as in code.
+
+### Know your audience
+
+The fact that Statey Studio can turn these executable diagrams into code is certainly powerful but what if much of your intended audience is less technical and will be viewing your state machines in the Studio? Some viewers may be more accustomed to reading some of the naming styles more than others, for example, **`Sentence case`** as opposed to **`snake_case`**. Users on the [Studio Pro Plan](https://stately.ai/docs/studio-pro-plan) can make use of [Version History](https://stately.ai/docs/versions) to create differently styled version of the same state machines and share them with different audiences either by [sharing the URL to a specific machine version](https://stately.ai/docs/url#copy-the-url-from-the-browser-address-bar) or by [exporting to markdown, Stories, tests, etc](https://stately.ai/docs/export-as-code).
```

Comment:
I often find non-coding users find snake case intimidating because it resembles code.